core/src/main/java/greencity/security/service/impl/GoogleSecurityServiceImpl.java

```diff
@@ -87,7 +85,7 @@ public SuccessSignInDto authenticate(String idToken) {
                     log.info("Google sign-in exist user - {}", user.getEmail());
                     return getSuccessSignInDto(user);
                 } catch (WrongEmailException e) {
-                    User user = createNewUser(email, familyName, givenName);
+                    User user = createNewUser(email, userName);
```

Comment:
Bad approach to create user in catch block. After catching an exception you should only log information if needed and handle correctly exception to give end user information what went wrong. Refactor repo not to throw exception when didn't found email in DB but just return null.